src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPath.cs

```diff
@@ -160,5 +162,14 @@ internal void Add(ODataPathSegment newSegment)
             ExceptionUtils.CheckArgumentNotNull(newSegment, "newSegment");
             this.segments.Add(newSegment);
         }
+
+        /// <summary>
+        /// Adds a range of segements to the current path
+        /// </summary>
+        /// <param name="oDataPath"></param>
+        public void AddRange(ODataPath oDataPath)

```

Comment:
>public [](start = 8, length = 6)

Should this be public or internal? Do we have customer scenarios for this?  If we start out making it internal we can make it public later, but we can't go the other way.

Also, if we change the public API we'll need to update the baseline in the E2E tests.